azure/meta.go

```diff
@@ -0,0 +1,130 @@
+package azure
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/hairyhenderson/gomplate/v4/env"
+)
+
+// defaultEndpoint is the IMDS endpoint at Azure
+// https://learn.microsoft.com/en-us/azure/virtual-machines/instance-metadata-service
+const defaultEndpoint = "http://169.254.169.254"
+
+var (
+	// co is a ClientOptions populated from the environment.
+	co ClientOptions
+	// coInit ensures that `co` is only set once.
+	coInit sync.Once
+)
+
+// ClientOptions contains various user-specifiable options for a MetaClient.
+type ClientOptions struct {
+	Timeout time.Duration
+}
+
+// GetClientOptions - Centralised reading of AZURE_TIMEOUT
+func GetClientOptions() ClientOptions {
+	coInit.Do(func() {
+		timeout := env.Getenv("AZURE_TIMEOUT")
+		if timeout == "" {
+			timeout = "500"
+		}
+
+		t, err := strconv.Atoi(timeout)
+		if err != nil {
+			panic(fmt.Errorf("invalid AZURE_TIMEOUT value '%s' - must be an integer: %w", timeout, err))
+		}
+
+		co.Timeout = time.Duration(t) * time.Millisecond
+	})
+	return co
+}
+
+// MetaClient is used to access metadata accessible via the Azure compute instance
+// metadata service version 1.
```

Comment:
```suggestion
// metadata service.
```

certainly not version 1...